src/main/kotlin/com/hibob/lambdas/LambdasExercise.kt

```diff
@@ -0,0 +1,119 @@
+package com.hibob.lambdas
+
+import java.time.LocalDate
+
+/*
+1. Make the Runner.init function more readable using Lambda with Receiver function usage
+2. Initiate the variable movie in line 14 with the function createGoodMovie()
+3. Implement pretty print using buildString function
+4. Make SpidermanMovieProduceActions nullable (if not so yet) and make the relevant adjustments
+*/
+
+fun main() {
+    val movie = SpidermanNoWayHome()
+    val runner = Runner(movie)
+    val success = runner.init()
+
+    printSuccessMessage(success)
+    println("Pretty print:\n${movie.prettyPrint()}")
+//    println("Json: ${movie.toJson()}")
+}
+
+fun printSuccessMessage(success: Boolean) {
+    if (success) {
+        println("The movie was successful")
+    } else {
+        println("The movie failed")
+    }
+}
+
+val createGoodMovie: () -> SpidermanNoWayHome = {SpidermanNoWayHome()}
+
+interface SpidermanMovieProduceActions {
+    fun signTobeyMaguire()
+    fun signAndrew()
+    fun signTom()
+    fun getVillains()
+    fun isThereLockdown(): Boolean
+    fun publish(): Boolean
+
+    val title: String
+    val airDate: LocalDate
+    val imdbRank: Double
+}
+
+class SpidermanNoWayHome() : SpidermanMovieProduceActions {
+
+    override val title: String = "Spiderman - No Way Home"
+    override val airDate: LocalDate = LocalDate.of(2021, 12, 16)
+    override val imdbRank: Double = 9.6
+
+    fun prettyPrint(): String {
+        val bs = buildString {
+            append("$title\n")
+            append("$airDate\n")
+            append("$imdbRank\n")
+        }
+        return bs
+    }
+
+    override fun signTobeyMaguire() {
+        //  Tobey signed!
```

Comment:
What about the empty implementations?

Reply:
I think that in the lesson you told us not to implement. How would you like us to implement those methods?

Reply:
I meant the json function that was commented- nothing too serious here - just print the commented line and you'll get the whole movie production when you'll run it :)